### DISCUSSION TITLE
Securing endpoints from an external url call

### DISCUSSION THREAD

**Discussion by glowkeeper** [directus/directus#4597] [Archive]
In https://github.com/directus/directus/discussions/4551, I asked how it was possible to secure custom endpoints? The answer was that "The current user info is accessible in req.accountability". 

However, when I try calling the custom endpoint from an external app' by setting Authorization: Bearer ${access_token} in the header (where access_token is got via `/auth/login`), my endpoint just sees this:
```
Accountability:  {
  user: null,
  role: null,
  admin: false,
  ip: '127.0.0.1',
  userAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 11_2_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.82 Safari/537.36'
}
```
Consequently, I never get beyond this:
```
    if ( ( req.accountability?.admin !== true ) ||
         ( req.accountability?.user !== true ) ) {

        console.error(config.uRLs.resetPass.url + " unknown user");
        return next(new ServiceUnavailableException("Unavailable"));
    }
```
Please forgive me - maybe I'm missing the point? I believe the access_token is in `req.header` - so maybe there's something I must do beforehand to create a valid accountability object form access_token?

**Comment by rijkvanzanten** [comment]
The authentication check is run on every endpoint, regardless of it being a custom endpoint:

https://github.com/directus/directus/blob/45a9f2a06cc1ebf55581a79d2f64b8dcbe576655/api/src/app.ts#L126

How are you sending the access token? 🤔

**Reply by glowkeeper** [reply]
I _believed_ it was like this:
```
     fetch(url, {
        method: 'POST',
        headers: {
            Authorization: `Bearer ${access_token}`
        }
      })
```
That should work, right?

If so, I reckon I know what the problem is - the code in question is via one of my junior developers, and I've just spotted it saying this:
```
    fetch(url, {
        method: 'POST',
        headers: {
            Authorization: `bearer ${access_token}`
        }
      })
```
...small case _b_ - could that be the issue? Apologies, if so...

**Reply by rijkvanzanten** [reply]
Yep, there you go :) simple typo likely throwing things off. Try it with Bearer capitalized and lemme know 👍🏻

**Reply by glowkeeper** [reply]
Yeah - smallcase _b_ was the issue - Grrrrrr - apologies for wasting your time...

**Reply by azharul-islam** [reply]
@rijkvanzanten Is there a way of calling a custom endpoint from inside the app using a Custom button (presentation field) and passing the token?

**Reply by rijkvanzanten** [reply]
@azharul-islam Yeah! See https://docs.directus.io/guides/modules/#accessing-the-api-from-within-your-extension

**Reply by Silbad** [reply]
The new url : https://docs.directus.io/extensions/modules.html#example-accessing-the-api-from-within-your-extension
